clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java

```diff
@@ -81,15 +81,15 @@ public void testFull() throws Exception {
         int appends = batchSize / msgSize;
         for (int i = 0; i < appends; i++) {
             // append to the first batch
-            accum.append(tp1, 0L, key, value, null, maxBlockTimeMs);
+            accum.append(tp1, 0L, key, value, -1L, null, maxBlockTimeMs);
```

Comment:
did you consider simply overloading `append` so that 1) we don't have to break existing APIs and 2) we minimize the number of changes in the PR?